app/kubernetes-ingress-controller/1.0.x/concepts/ingress-versions.md

```diff
@@ -0,0 +1,164 @@
+---
+title: Ingress v1 and v1beta1 differences
+---
+
+## Introduction
+
+Kubernetes 1.18 introduced a new v1 API for the [Ingress resource][kubernetes-ingress-doc].
+It standardizes common practices and clarifies implementation requirements that
+were previously up to individual controller vendors. This document covers those
+changes as they relate to Kong Ingress Controller and provides example
+equivalent v1beta1 and v1 resources for comparison.
+
+## Paths
+
+Both Ingress v1beta1 and v1 HTTP rules require a path, which represents a [URI
+path][uri-rfc-paths]. Although v1beta1 had specified that paths were [POSIX
+regular expressions][posix-regex] and enforced this, but in practice most
+controllers used other other implementations that did not match the
+specification. v1 seeks to reduce confusion by introducing several i[path
+types][path-types] and lifting restrictions on regular expression grammars used
+by controllers.
+
+### v1beta1
+
+The controller passes paths directly to Kong and relies on its [path handling
+logic][kong-path]. The Kong proxy treats paths as a prefix unless they include
+characters [not allowed in RFC 3986 paths][uri-rfc-paths], in which case the
+proxy assumes they are a regular expression, and does not treat slashes as
+special characters. For example, the prefix `/foo` can match any of the
+following:
+
+```
+/foo
+/foo/
+/foobar
+/foo/bar
+```
+
+### v1
+
+Although v1 Ingresses provide path types with more clearly-defined logic, the
+controller must still create Kong routes and work within the Kong proxy's
+routing logic. As such, the controller translates Ingress rule paths to create
+Kong routes that match the specification:
+
+#### Exact
+
+To handle exact matches, the controller creates a Kong route with a regular
+expression that matches the rule path only, e.g. an exact rule for `/foo` in an
+Ingress translates to a Kong route with a `/foo$` regular expression path.
+
+#### Prefix
+
+To handle prefix matches, the controller creates a Kong route with two path
+criteria, e.g. `/foo` will create a route with a `/foo$` regular expression and
+`/foo/` plain path.
+
+#### ImplementationSpecific
+
+The controller leaves ImplementationSpecific path rules entirely up to the Kong
+router. It creates a route with the exact same path string as the Ingress rule.
+
+## Ingress class
+
+[Ingress class][ingress-class] indicates which resources an ingress controller
+should process. It provides a means to separate out configuration intended for
+other controllers or other instances of the Kong Ingress Controller.
+
+In v1beta1, ingress class was handled informally using
+`kubernetes.io/ingress.class` [annotations][deprecated-annotation]. v1
+introduces a new [IngressClass resource][ingress-class-api] which provides
+richer information about the controller. v1 Ingresses are bound to a class via
+their `ingressClassName` field.
+
+For example, consider this v1beta1 Ingress:
+
+```
+apiVersion: extensions/v1beta1
```

Comment:
```suggestion
apiVersion: networking.k8s.io/v1beta1
```

Let's explain the migration one step at a time - from `networking.k8s.io/v1beta1` to `networking.k8s.io/v1`. To avoid confusion, we should probably avoid mixing the two in one doc as well.

Particularly so because migration from `extensions/v1beta1` to `networking.k8s.io/v1beta1` is a no-op.